tests/backup/backup_azure_credential_test.go

```diff
@@ -0,0 +1,231 @@
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s with mandatory fields in azure credentials", azureBackupLocationNameWithMandatoryFields))
+			}
+		})
+
+		Step("Registering azure application clusters using cloud credential having all fields", func() {
+			log.InfoD("Registering azure application clusters using cloud credential having all fields")
+			err = AddAzureApplicationClusters(BackupOrgID, azureCredNameWithAllFields, credUidWithAllFields, ctx)
+			dash.VerifyFatal(err, nil, "Adding source and destination cluster using cloud credential having all fields")
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, BackupOrgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Taking backup of applications using backup location with cloud credential having only mandatory parameters", func() {
+			log.InfoD("Taking backup of applications using backup location with cloud credential having only mandatory parameters")
+			backupNameWithBkpLocationHavingMandatoryParameters = fmt.Sprintf("backup-with-bkp-loc-having-mandatory-params-%v", RandomString(5))
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, appNamespaces)
+			err = CreateBackupWithValidation(ctx, backupNameWithBkpLocationHavingMandatoryParameters, SourceClusterName, azureBackupLocationNameWithMandatoryFields, backupLocationMandatoryFieldsUID, appContextsToBackup, nil, BackupOrgID, sourceClusterUid, "", "", "", "")
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupNameWithBkpLocationHavingMandatoryParameters))
+		})
+
+		Step("Restoring the backed up application from backup location with cloud credential having only mandatory parameters", func() {
+			log.InfoD("Restoring the backed up application from backup location with cloud credential having only mandatory parameters")
+			restoreNameWithBkpLocationHavingMandatoryParameters = fmt.Sprintf("restore-%s-%v", backupNameWithBkpLocationHavingMandatoryParameters, RandomString(5))
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, appNamespaces)
+			err = CreateRestoreWithValidation(ctx, restoreNameWithBkpLocationHavingMandatoryParameters, backupNameWithBkpLocationHavingMandatoryParameters, make(map[string]string), make(map[string]string), DestinationClusterName, BackupOrgID, appContextsToBackup)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Validating restore-%s", restoreNameWithBkpLocationHavingMandatoryParameters))
+		})
+
+		Step("Removing the added application cluster and registering new azure application clusters using cloud credential having only mandatory fields", func() {
+			log.InfoD("Deleting registered clusters for admin context")
+			err = DeleteCluster(SourceClusterName, BackupOrgID, ctx, true)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
```

Comment:
should we have wait for cluster deletion here and not fail safely .

Reply:
corrected